tacker/common/utils.py

```diff
@@ -299,3 +300,11 @@ def cpu_count():
         return multiprocessing.cpu_count()
     except NotImplementedError:
         return 1
+
+
+def is_auth_uri_v3(auth_url):
+    return (re.match('.+v3/?$', auth_url) is not None)
+
+
+def auth_uri_unspecified_version(auth_url):
```

Comment:
Maybe adding the whole logic into this block makes it easier to use:

```
def get_correct_auth_url(auth_url):
    if re.match('.+(v3/?|v2\.0/?)$', auth_url) is None:
        auth_url = '{0}/v2.0'.format(auth_url)

    return auth_url
```

and in all other files we just call it with:

```
auth_url = common_utils.get_correct_auth_url(cfg.CONF.keystone_authtoken.auth_uri)
```